### PR TITLE
Resources documented with JSDoc

### PR DESCRIPTION
**Pull request by SampaioLeal**

### CODE REVIEW

File: src/EntityStore.ts [hunk 1]
```diff
@@ -14,22 +22,45 @@ export default class EntityStore {
...
   }
 
+  /**
+   * The loading state of the store. If it's true, the store is fetching data.
+   */
   loading = false;
+  /**
+   * The data that represents an entity.
+   */
   data: unknown | null = null;
```
Comment (devzgabriel) [inline src/EntityStore.ts:32]: couldn't typing be dynamic by generics?

**Review by devzgabriel — APPROVED** [review]

File: src/EntityStore.ts (hunk 1 above)
Comment (SampaioLeal) [reply src/EntityStore.ts:32]: in update function we have a patch request with a `this.setData` call using `response.data` that is `unknown`
if i put a `T` in Axios response we are assuming that the patch or put endpoint will respond with the same entity structure of the GET one

Comment (SampaioLeal) [reply src/EntityStore.ts:32]: can i proceed?

Comment (devzgabriel) [reply src/EntityStore.ts:32]: Yes, PR confirmed

**Pull request merged**